tree_in_context/src/utility/util.egg

```diff
@@ -35,3 +35,37 @@
        (= len1 (tuple-length expr1))
        (= ith (tuple-ith expr2 i)))
        ((union (tuple-ith (Concat ord expr1 expr2) (+ len1 i)) ith)) :ruleset always-run)
+
+(function tuple-remove-fst (Expr) Expr :unextractable)
+(rule ((tuple-remove-fst tup)
+       (= tup (Concat ord (Single expr1) expr2)))
+      ((union (tuple-remove-fst tup) expr2)) :ruleset always-run)
+
+(rule ((tuple-remove-fst tup)
+       (= tup (Concat ord expr1 (Single expr2)))
+       (HasType expr1 (TupleT (TNil))))
+      ((union (tuple-remove-fst tup) expr1)) :ruleset always-run)
+
+(rule ((tuple-remove-fst tup)
+       (= tup (Concat ord expr1 expr2))
+       (HasType expr1 (TupleT (TCons ty tl))))
+      ((union (tuple-remove-fst tup) (Concat ord (tuple-remove-fst expr1) expr2))) :ruleset always-run)
+
+(rule ((tuple-remove-fst tup)
+       (= tup (Concat ord expr1 expr2))
+       (HasType expr1 (TupleT (TNil))))
+      ((union (tuple-remove-fst tup) (Concat ord expr1 (tuple-remove-fst expr2)))) :ruleset always-run)
+
+(function DoWhile-outputs-ith (Expr i64) Expr :unextractable)
```

Comment:
Are these helpers necessary? They create extra rows in the database.
Maybe the rule writer should just use tuple-ith